### PR TITLE
Configure MySQL Database

### PR DESCRIPTION
**Pull request by halogenandtoast**

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -7,12 +7,16 @@ import Web.Spock.Config
...
-data AppSettings = AppSettings { appPort :: Int }
+data AppSettings = AppSettings { appPort :: Int
+                               , appDatabaseConf :: MySQLConf
+                               }
 
 instance FromJSON AppSettings where
   parseJSON = withObject "AppSettings" $ \o -> do
     appPort <- o .: "port"
+    appDatabaseConf <- o .: "database"
     return AppSettings{..}
```
Comment (pauchan) [inline app/Main.hs]: (question) would that be possible to reverse the order here and drop `do` syntax similar to what they do in [Argo](https://github.com/thoughtbot/Argo):
```
curry(User.init)
      <^> json <| "id"
      <*> json <| "name"
```

Comment (pauchan) [reply app/Main.hs]: Also, why `o` ? Can we rename it to `json` ?

**Review by pauchan — APPROVED** [review]
LGTM! 🚀

File: settings.yml [hunk 2]
```diff
@@ -1 +1,8 @@
 port: "_env:PORT:3000"
+database:
+  user: "_env:MSUSER:root"
+  password: "_env:MSPASS:"
+  host: "_env:MSHOST:localhost"
+  port: "_env:MSPORT:3306"
+  database: "_env:MSDATABASE:mirin"
+  poolsize: "_env:MSPOOLSIZE:5"
```
Comment (JuanitoFatas) [inline settings.yml]: What do you think about following the naming convention employs in our web application and prefix them with the application name?

```
user: "_env:MIRIN_DATABASE_USER:root"
```

```
MIRIN_DATABASE_USER
MIRIN_DATABASE_PASSWORD
MIRIN_DATABASE_HOST
MIRIN_DATABASE_PORT
MIRIN_DATABASE_NAME
MIRIN_DATABASE_POOLSIZE
```

File: app/Main.hs (hunk 1 above)
Comment (halogenandtoast) [reply app/Main.hs]: o is for Object (it is no longer json). Also the Haskell way is to use short names when types can tell you what you want to know. We can break that convention, but most examples are going to look like what I've done here for parsing JSON.

**Review by JuanitoFatas — APPROVED** [review]

File: settings.yml (hunk 2 above)
Comment (eagletmt) [reply settings.yml]: +1. More descriptive names like `DATABASE_USER` or `MIRIN_DATABASE_USER` looks better.

**Review by eagletmt — APPROVED** [review]

**Review by eagletmt — APPROVED** [review]

**Comment by JuanitoFatas** [comment]
🐑

**Pull request merged**